tests/vlan_ping/test_vlan_ping.py

```diff
@@ -0,0 +1,157 @@
+
+        else:
+            logger.debug("unknown IP version")
+
+
+@pytest.fixture(scope='module')
+def vlan_ping_setup(duthosts, rand_one_dut_hostname, ptfhost, nbrhosts):
+    """
+    Setup:      adds ipv4 and ipv6 address on ptf hosts and routes for VM
+    Teardown:   deletes ipv4 and ipv6 address on ptf hosts and removes routes to VM. Also removes residual static arp entries from tests
+    """
+    try:
+        k, v = random.choice(nbrhosts.items())
+        vm_host = v['host']
+        vm_host_ip = (v['conf']['interfaces']['Port-Channel1']['ipv4']).decode('utf-8')
+        vm_host_network = ipaddress.IPv4Interface(format(vm_host_ip)).network
+        duthost = duthosts[rand_one_dut_hostname]
+        my_cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+        ptfhost_info = {}
+
+        # getting the ipv4, ipv6 and vlan id of a vlan in DUT with 2 or more vlan members
+        for k, v in my_cfg_facts['VLAN'].items():
+            vlanid = v['vlanid']
+            if len(my_cfg_facts['VLAN_MEMBER']['Vlan' + vlanid]) >= 2:
+                for addr in my_cfg_facts['VLAN_INTERFACE']['Vlan' + vlanid]:
+                    if addr.find(':') == -1:
+                        ip4 = addr
+                    else:
+                        ip6 = addr
+                break  # need only 1 vlan details
+            else:
+                continue
+
+        # ip prefixes of the vlan
+        vlan_ip_address_v4 = ipaddress.IPv4Interface(ip4).ip
+        vlan_ip_prefixlen_v4 = ipaddress.IPv4Interface(ip4).network.prefixlen
+        vlan_ip_prefixlen_v6 = ipaddress.IPv6Interface(ip6).network.prefixlen
+
+        # selecting 2 random vlan members of DUT
+        exclude_member = ["Ethernet0", "Ethernet1"]
+        vlan_member_list = [ele for ele in my_cfg_facts['VLAN_MEMBER']['Vlan' + vlanid].keys() if ele not in exclude_member]
+        rand_vlan_member_list = random.sample(vlan_member_list, 2)
+
+        # getting port index, mac, ipv4 and ipv6 of ptf ports into a dict
+        for member in rand_vlan_member_list:
+            ptfhost_info[member] = {}
+            ptfhost_info[member]["Vlanid"] = vlanid
+            ptfhost_info[member]["port_index"] = my_cfg_facts['port_index_map'][member]
+            ptfhost_info[member]["mac"] = (ptfhost.shell(
+                "ifconfig eth%d | grep -o -E '([[:xdigit:]]{1,2}:){5}[[:xdigit:]]{1,2}'" % ptfhost_info[member][
+                    "port_index"]))['stdout']
+            ptfhost_info[member]["ipv4"] = str(
+                ipaddress.IPv4Interface(ip4).network[ptfhost_info[member]["port_index"]])
+            ptfhost_info[member]["ipv6"] = str(
+                ipaddress.IPv6Interface(ip6).network[ptfhost_info[member]["port_index"]])
+            # add ipv4 and v6 to randomly selected ports in ptf
+            logger.info("adding ip to ptf ports")
+            logger.debug("adding ipv4 of ptf port %d " % ptfhost_info[member]['port_index'])
```

Comment:
Configuring addresses in same network in same VM(ptf_host) on different interfaces is not correct.
=========================================
root@584f99893660:~# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.5.208.1      0.0.0.0         UG    0      0        0 mgmt
10.0.0.56       192.168.0.1     255.255.255.254 UG    0      0        0 eth45
10.5.208.0      0.0.0.0         255.255.255.0   U     0      0        0 mgmt
10.10.246.0     0.0.0.0         255.255.255.0   U     0      0        0 backplane
192.168.0.0     0.0.0.0         255.255.248.0   U     0      0        0 eth45
192.168.0.0     0.0.0.0         255.255.248.0   U     0      0        0 eth41
root@584f99893660:~#
root@584f99893660:~#
root@584f99893660:~# ifconfig eth45
eth45: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9000
        inet 192.168.0.45  netmask 255.255.248.0  broadcast 0.0.0.0
        inet6 fe80::ae16:2dff:fe76:202d  prefixlen 64  scopeid 0x20<link>
        inet6 fc02:1000::2d  prefixlen 64  scopeid 0x0<global>
        ether ac:16:2d:76:20:2d  txqueuelen 1000  (Ethernet)
        RX packets 81631  bytes 10058412 (9.5 MiB)
        RX errors 0  dropped 117  overruns 0  frame 0
        TX packets 15367  bytes 1278274 (1.2 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

root@584f99893660:~# ifconfig eth41
eth41: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9000
        inet 192.168.0.41  netmask 255.255.248.0  broadcast 0.0.0.0
        inet6 fc02:1000::29  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::ae16:2dff:fe76:2029  prefixlen 64  scopeid 0x20<link>
        ether ac:16:2d:76:20:29  txqueuelen 1000  (Ethernet)
        RX packets 83003  bytes 10134654 (9.6 MiB)
        RX errors 0  dropped 116  overruns 0  frame 0
        TX packets 17342  bytes 1528310 (1.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        
        In this setup, only one IP will ping at a time from DUT and it will lead to random failures